### DISCUSSION TITLE
How to ignore cert errors within Hijack Requests?

### DISCUSSION THREAD

**Discussion by CyberStryk** [go-rod/rod#988] [Q&A]
Are you able to ignore certificate errors inside HijackRequest? I am unable to ignore certificate errors while hijacking requests with both browser.HijackRequests() and page.HijackRequests()  

I set my browser to IgnoreCertErrors and I am able to properly navigate normally without any issues.When I try to navigate using hijack I get Panic certificate errors. 

```go
func HijackViaPage() {
	browser := rod.New().MustConnect()
	browser.MustIgnoreCertErrors(true)
	page := browser.MustPage("")

	// THIS WORKS, No issues.
	page.MustNavigate("https://127.0.0.1/")

	router := page.HijackRequests()

	router.MustAdd("*", func(ctx *rod.Hijack) {

		ctx.Request.Req().Host = "test.host"
		ctx.Request.Req().Header.Set("Accept-Language", "test-language")

		// PANIC: Certificate issue.
		ctx.MustLoadResponse()
	})

	go router.Run()

	// PANIC: Certificate issue.
	page.MustNavigate("https://127.0.0.1/")

}
```
Error:
```bash
        /home/user/go/pkg/mod/github.com/go-rod/rod@v0.114.0/lib/utils/utils.go:68 +0x1d
main.HijackViaPage.New.(*Browser).WithPanic.genE.func2({0xc0002befb0?, 0xf?, 0xc0001aeb01?})
        /home/user/go/pkg/mod/github.com/go-rod/rod@v0.114.0/must.go:36 +0x5a
github.com/go-rod/rod.(*Hijack).MustLoadResponse(0xc0001a2300)
        /home/user/go/pkg/mod/github.com/go-rod/rod@v0.114.0/must.go:1121 +0x8f
main.HijackViaPage.func1(0xc0001a2300)
        /home/userdevelopment/vhost-screenshot/vhost-screenshot.go:21 +0x85
github.com/go-rod/rod.(*HijackRouter).initEvents.func1.1()
        /home/user/go/pkg/mod/github.com/go-rod/rod@v0.114.0/hijack.go:77 +0x122
created by github.com/go-rod/rod.(*HijackRouter).initEvents.func1 in goroutine 9
        /home/user/go/pkg/mod/github.com/go-rod/rod@v0.114.0/hijack.go:70 +0xaa
```

I have also tried modifying every function in hijack.go to set browser to IgnoreCertErrors, but that did not seem to work. 

Is it an issue with my MustAdd context? Or is this an upstream issue with Chrome?

**Comment by CyberStryk** [comment]
Opening issue to track.

**Reply by xavier268** [reply]
Same issue here. Hijack generates "tls: failed to verify certificate: X509; certificate signed with unknown authority". I was able to skip certificates control both for direct https GET requests, and with browser.IgnoreCertErrors(true), but does not work for hijack !

**Reply by xavier268** [reply]
Actually, I fixed it for my use case by using a specific tls transport with the hijack. 

I replaced : 

      router.MustAdd("*/api/affaire-liste", func(ctx *rod.Hijack) {
		ctx.MustLoadResponse()
		chanAffList <- ctx.Response.Body()
		close(chanAffList)
	})
	defer router.MustStop()

By : 

     router.MustAdd("*/api/affaire-liste", func(ctx *rod.Hijack) {
		// ctx.MustLoadResponse()
		ctx.LoadResponse(
			&http.Client{
				Transport: &http.Transport{
					TLSClientConfig: &tls.Config{
						InsecureSkipVerify: true}}},
			true)
		chanAffList <- ctx.Response.Body()
		close(chanAffList)
	})
	defer router.MustStop()
